[pruner] refactor `ActivationReconstruction` forward hooks

kazhou opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63178
* __->__ #63158

Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

kazhou committed:
[pruner] refactor `ActivationReconstruction` forward hooks

Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

raghuramank100 reviewed (approved):
Is the name ActivationReconstruction appropriate given that we are applying it to weights?

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] refactor `ActivationReconstruction` forward hooks"


Combined functionality for `ActivationReconstruction` for both Linear and Conv2d in one class. The only difference between the old classes was the size and indexing of the reconstructed tensor -- that logic can be generalized by iterating over the size of `output`.

Differential Revision: [D30282765](https://our.internmc.facebook.com/intern/diff/D30282765/)

[ghstack-poisoned]
